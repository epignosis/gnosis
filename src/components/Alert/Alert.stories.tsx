import React from "react";
import { Story } from "@storybook/react";
import Alert, { AlertProps } from "./Alert";

export default {
  component: Alert,
  title: "Components/Alert",
  argTypes: {
    type: {
      control: {
        type: "select",
        options: ["info", "danger", "success", "warning"],
      },
    },
    icon: {
      control: false,
    },
  },
  args: {
    type: "info",
    children: "Hi I am an Alert component!",
  },
};

const Template: Story<AlertProps> = (args) => <Alert {...args} />;

export const Default = Template.bind({});

export const WithCloseBtn = Template.bind({});

WithCloseBtn.argTypes = {
  onClose: { action: "onClose" },
};
