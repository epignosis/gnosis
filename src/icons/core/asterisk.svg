<svg id="ic_advanced_search" data-name="ic_advanced search" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
  <path id="Path_2112" data-name="Path 2112" d="M22.468-4.255,14.25-9l8.218-4.744a.563.563,0,0,0,.206-.768l-.562-.974a.563.563,0,0,0-.768-.206l-8.218,4.744v-9.489A.563.563,0,0,0,12.562-21H11.437a.563.563,0,0,0-.562.563v9.489L2.657-15.693a.563.563,0,0,0-.768.206l-.562.974a.563.563,0,0,0,.206.768L9.75-9,1.532-4.255a.563.563,0,0,0-.206.768l.562.974a.563.563,0,0,0,.768.206l8.218-4.744V2.437A.563.563,0,0,0,11.437,3h1.125a.563.563,0,0,0,.562-.562V-7.051l8.218,4.744a.563.563,0,0,0,.768-.206l.563-.974A.563.563,0,0,0,22.468-4.255Z" transform="translate(0 21)"/>
  <rect id="Rectangle_1183" data-name="Rectangle 1183" width="24" height="24" transform="translate(0 0)" fill="none"/>
</svg>
