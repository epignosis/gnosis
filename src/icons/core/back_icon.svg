<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 25 25">
  <g id="icon_goback" transform="translate(-1097 -1370)">
    <rect id="Rectangle_1063" data-name="Rectangle 1063" width="25" height="25" transform="translate(1097 1370)" fill="#fff"/>
    <path id="Path_132" data-name="Path 132" d="M4.765-1.616l.613-.613a.375.375,0,0,0-.005-.536L2.851-5.187H13.625A.375.375,0,0,0,14-5.562v-.875a.375.375,0,0,0-.375-.375H2.851L5.373-9.235a.375.375,0,0,0,.005-.536l-.613-.613a.375.375,0,0,0-.53,0L.116-6.265a.375.375,0,0,0,0,.53L4.235-1.616A.375.375,0,0,0,4.765-1.616Z" transform="translate(1103 1388)"/>
  </g>
</svg>
