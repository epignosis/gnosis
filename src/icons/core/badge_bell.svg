<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 95 95">
  <g id="icon_dontmiss" transform="translate(-126.242)">
    <circle id="Ellipse_220" data-name="Ellipse 220" cx="47.5" cy="47.5" r="47.5" transform="translate(126.242)" fill="#0e59bb"/>
    <g id="Group_13104" data-name="Group 13104" transform="translate(145 19)">
      <path id="Path_2101" data-name="Path 2101" d="M14.341-16.46C12.653-18.274,9.493-21,9.493-29.945A13.8,13.8,0,0,0-1.689-43.507v-1.821a2.8,2.8,0,0,0-2.8-2.8,2.8,2.8,0,0,0-2.8,2.8v1.821A13.8,13.8,0,0,0-18.462-29.945c0,8.941-3.16,11.671-4.848,13.485a2.731,2.731,0,0,0-.753,1.9,2.8,2.8,0,0,0,2.806,2.8H12.288a2.8,2.8,0,0,0,2.806-2.8A2.73,2.73,0,0,0,14.341-16.46Zm-32.5.5c1.855-2.445,3.882-6.5,3.892-13.934,0-.017-.005-.033-.005-.051a9.789,9.789,0,0,1,9.789-9.789A9.789,9.789,0,0,1,5.3-29.945c0,.017-.005.033-.005.051.01,7.438,2.037,11.49,3.892,13.934ZM-4.484-3.375A5.593,5.593,0,0,0,1.107-8.969H-10.075A5.593,5.593,0,0,0-4.484-3.375Z" transform="translate(33.063 54.125)" fill="#fff"/>
      <rect id="Rectangle_1633" data-name="Rectangle 1633" width="57" height="57" transform="translate(0.242)" fill="none"/>
    </g>
  </g>
</svg>
