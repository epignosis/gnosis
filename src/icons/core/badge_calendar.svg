<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 95 95">
  <g id="icon_today" transform="translate(-1390 -977)">
    <circle id="Ellipse_221" data-name="Ellipse 221" cx="47.5" cy="47.5" r="47.5" transform="translate(1390 977)" fill="#0e59bb"/>
    <path id="Path_2102" data-name="Path 2102" d="M10.117-42.656h-4.1V-47.1A1.028,1.028,0,0,0,4.99-48.125H1.572A1.028,1.028,0,0,0,.547-47.1v4.443H-10.391V-47.1a1.028,1.028,0,0,0-1.025-1.025h-3.418A1.028,1.028,0,0,0-15.859-47.1v4.443h-4.1a4.1,4.1,0,0,0-4.1,4.1V-8.477a4.1,4.1,0,0,0,4.1,4.1H10.117a4.1,4.1,0,0,0,4.1-4.1V-38.555A4.1,4.1,0,0,0,10.117-42.656ZM9.6-8.477H-19.448a.514.514,0,0,1-.513-.513V-34.453H10.117V-8.989A.514.514,0,0,1,9.6-8.477Z" transform="translate(1442.063 1050.125)" fill="#fff"/>
    <rect id="Rectangle_1634" data-name="Rectangle 1634" width="57" height="57" transform="translate(1409 996)" fill="none"/>
  </g>
</svg>
