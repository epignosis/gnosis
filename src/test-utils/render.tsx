import React, { FC, ReactElement, ComponentType } from "react";
import { render, RenderResult, RenderOptions } from "@testing-library/react";
import ThemeProvider from "../components/ThemeProvider/ThemeProvider";

const AllTheProviders: FC = ({ children }) => <ThemeProvider>{children}</ThemeProvider>;

const customRender = (ui: ReactElement, options: RenderOptions): RenderResult =>
  render(ui, { wrapper: AllTheProviders as ComponentType, ...options });

export * from "@testing-library/react";

export { customRender as render };
