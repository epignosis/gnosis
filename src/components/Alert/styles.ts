import { css, Theme, SerializedStyles } from "@emotion/react";
import { AlertType } from "./Alert";
import { hexToRGBA } from "@theme/default/colors";
import { mq } from "@theme/utils/breakpoints";

export const container = ({ alert }: Theme, { type }: { type: AlertType }): SerializedStyles => {
  return css`
    width: 100%;
    position: relative;
    display: flex;
    align-items: center;
    background-color: ${alert[type].background};
    color: ${alert[type].color};
    border-radius: 5px;
    padding: 2rem 3.5rem 2rem 1.75rem;

    .icon {
      display: none;
      margin-right: 1.5rem;

      ${mq["md"]} {
        display: block;
      }
    }

    .close-btn {
      position: absolute;
      height: 100%;
      right: 0.5rem;
      cursor: pointer;
      color: ${hexToRGBA(alert.closeBtnColor, 0.25)};
      background: transparent;
      padding: 0;
      border: 0;

      ${mq["xl"]} {
        right: 1.5rem;
      }
    }
  `;
};
