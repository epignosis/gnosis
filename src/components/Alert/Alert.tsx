import React, { FC } from "react";
import { SerializedStyles } from "@emotion/react";
import { WarningSVG, DangerSVG, InfoSVG, SuccessSVG, CloseCircledSVG } from "../../icons/";
import { container } from "./styles";
import { IconType } from "types/common";

export type AlertType = "info" | "danger" | "success" | "warning";

export type AlertProps = React.HTMLAttributes<HTMLDivElement> & {
  type: AlertType;
  onClose?: () => void;
  icon?: IconType;
};

const ICONS = {
  info: <InfoSVG height={56} />,
  danger: <DangerSVG height={56} />,
  success: <SuccessSVG height={56} />,
  warning: <WarningSVG height={56} />,
};

const Alert: FC<AlertProps> = ({ type, children, onClose, icon: Icon, ...rest }) => {
  return (
    <div css={(theme): SerializedStyles => container(theme, { type })} {...rest}>
      {Icon ? (
        <Icon height={56} className="icon" data-testid="icon" />
      ) : (
        <div className="icon" data-testid="icon">
          {ICONS[type]}
        </div>
      )}
      <article>{children}</article>
      {Boolean(onClose) && (
        <button aria-label="Close alert" onClick={onClose} className="close-btn">
          <CloseCircledSVG height={32} />
        </button>
      )}
    </div>
  );
};

export default Alert;
