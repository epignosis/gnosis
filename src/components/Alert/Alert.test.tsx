import React from "react";
import faker from "faker";
import userEvent from "@testing-library/user-event";
import Alert from "./Alert";
import { render, screen } from "@test-utils/render";
import { Heading } from "@components";
import { ScrollRegularSVG } from "@icons/core";

describe("<Alert>", () => {
  const mockedOnClose = jest.fn();

  it("renders correctly", () => {
    const heading = faker.lorem.word();
    const paragraphTxt = faker.lorem.paragraph();

    render(
      <Alert type="info" onClose={mockedOnClose}>
        <Heading as="h3" size="md">
          {heading}
        </Heading>
        <p>{paragraphTxt}</p>
      </Alert>,
    );
    const title = screen.getByText(heading);
    const paragraph = screen.getByText(paragraphTxt);
    const closeLink = screen.getByRole("button");
    const icon = screen.getByTestId(/icon/i);

    expect(title).toHaveTextContent(heading);
    expect(paragraph).toHaveTextContent(paragraphTxt);
    userEvent.click(closeLink);
    expect(mockedOnClose).toHaveBeenCalledTimes(1);
    expect(icon).toBeInTheDocument();
  });

  it("renders correctly with custom icon", () => {
    const paragraphTxt = faker.lorem.paragraph();

    render(
      <Alert type="info" onClose={mockedOnClose} icon={ScrollRegularSVG}>
        {paragraphTxt}
      </Alert>,
    );

    const icon = screen.getByTestId(/icon/i);

    expect(icon).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const { container } = render(
      <Alert type="info" onClose={mockedOnClose}>
        <Heading as="h3" size="md">
          Welcome
        </Heading>
        <p>My content!</p>
      </Alert>,
    );

    expect(container).toMatchSnapshot();
  });

  it("matches snapshot with custom icon", () => {
    const { container } = render(
      <Alert type="info" onClose={mockedOnClose} icon={ScrollRegularSVG}>
        <p>My content!</p>
      </Alert>,
    );

    expect(container).toMatchSnapshot();
  });
});
