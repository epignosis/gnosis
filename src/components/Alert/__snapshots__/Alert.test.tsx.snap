// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alert> matches snapshot 1`] = `
<div>
  <div
    class="css-k5zh3p-container-container"
  >
    <div
      class="icon"
      data-testid="icon"
    >
      <svg-file-stub
        height="56"
      />
    </div>
    <article>
      <h3
        class="css-1iyd91n-headingContainer-headingContainer"
      >
        Welcome
      </h3>
      <p>
        My content!
      </p>
    </article>
    <button
      aria-label="Close alert"
      class="close-btn"
    >
      <svg-file-stub
        height="32"
      />
    </button>
  </div>
</div>
`;

exports[`<Alert> matches snapshot with custom icon 1`] = `
<div>
  <div
    class="css-k5zh3p-container-container"
  >
    <svg-file-stub
      classname="icon"
      data-testid="icon"
      height="56"
    />
    <article>
      <p>
        My content!
      </p>
    </article>
    <button
      aria-label="Close alert"
      class="close-btn"
    >
      <svg-file-stub
        height="32"
      />
    </button>
  </div>
</div>
`;
